SofaKernel/framework/sofa/simulation/MechanicalOperations.cpp

```diff
@@ -511,7 +511,8 @@ void MechanicalOperations::addMBK_ToMatrix(const sofa::core::behavior::MultiMatr
     if (matrix != NULL)
     {
         //std::cout << "MechanicalAddMBK_ToMatrixVisitor "<< mFact << " " << bFact << " " << kFact << " " << offset << std::endl;
-        executeVisitor( MechanicalAddMBK_ToMatrixVisitor(&mparams, matrix) );
+        executeVisitor( MechanicalAddMBK_ToMatrixVisitor(&mparams /* PARAMS FIRST */, matrix) );
+        executeVisitor( MechanicalAddProjectiveConstraint_ToMatrixVisitor(&mparams /* PARAMS FIRST */, matrix) );
```

Comment:
HARDLY disagree about always applying the projection matrix in the addMBK visitor.
It completely depends on the numerical solver you are using.
After projection MBK is no longer positive definite.

+ mappings can also add geometric stiffness that would not be projected.

Did you make modifications in numerical solvers by considering the MBK matrix as always projected?